targets/TARGET_Silicon_Labs/TARGET_EFM32/pwmout_api.c

```diff
@@ -334,24 +341,34 @@ void pwmout_period_us(pwmout_t *obj, int us)
     pwmout_period(obj, us / 1000000.0f);
 }
 
+int pwmout_read_period_us(pwmout_t *obj)
+{
+    return TIMER_TopGet(PWM_TIMER) / (REFERENCE_FREQUENCY * 1000000);
```

Comment:
That's even worse, this will almost guaranteed return zero.

What you'd want:
` return (TIMER_TopGet(PWM_TIMER) * 1000000) / REFERENCE_FREQUENCY;`

Reply:
sorry you are completely right